app/views/orders/_sales_order.html.erb

```diff
@@ -13,7 +13,7 @@
               <%= content_tag_for(:tr, item, id: item.product.id, class: "product order-item-row", data: {id: item.product.id}) do %>
                 <td class="name"><%= item.product.name %></td>
                 <td class="quantity math"><%= item.decorate.quantity_with_unit %></td>
-                <td class="price math"><%= number_to_currency item.unit_price %></td>
+                <td class="price math"><%= number_to_currency item.unit_price.sale_price %></td>
```

Comment:
@thermistor why does the price show correctly on the first view of the cart but not when coming back from validation error?

Reply:
It's because there are different `_sales_order.html.erb` partials that are used depending on what action the page is rendered from (see also `carts/_sales_order.html.erb`). The issue this fixes only shows up when there is a validation error.

Reply:
I guess my implied question is, why is there a different partial when viewing card versus coming back from validation? It seems like an egregious code duplication. Is it a bit tricky to decouple (I don't want to delay this, just hoping to learn a bit)?

Reply:
Yeah, not sure entirely why there are two partials. I didn't dig too deep, I couldn't tell if it was just an oversight, not realizing that a partial with the same name would get rendered in the case of a validation error when posting to  `/orders` or if it was meant to have different functionality in the error case.

Reply:
I dug a little into this, I'm not entirely sure of the all the rationale behind having a `Price` object ( though `Order::UnitPriceLogic` seems to explain some of the wackiness). This solution _looks_ legit though, but Rob's question about duplication remains..